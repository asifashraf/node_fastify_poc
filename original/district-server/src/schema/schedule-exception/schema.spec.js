const { fetchGraphQL, getFirstId } = require('../../lib/test-util');
const { scheduleExceptionsDetails } = require('../../lib/test-fragments');
const {
  costa1,
} = require('../../../database/seeds/development').brandLocations;

test('brand locations can resolve scheduleException', async () => {
  const query = `{
    brandLocation(id: "${costa1.id}") {
      scheduleExceptions { ${scheduleExceptionsDetails} }
    }
}  `;
  // openInGraphiql(query);
  const result = await fetchGraphQL(query);
  expect(result).toMatchSnapshot();
});

test('scheduleExceptions can be added', async () => {
  const { id: brandLocationId } = await getFirstId('brand_locations');

  const query = `mutation { brandLocationScheduleExceptionSave(scheduleExceptions: [
    {
      brandLocationId: "${brandLocationId}"
		  startTime:"2021-11-16T01:20:45+01:00"
		  endTime:"2022-11-16T12:20:14+01:00"
      isClosed:false
      isDeliveryClosed:false
      isExpressDeliveryClosed:false
    },
    {
      brandLocationId: "${brandLocationId}"
		  startTime:"2021-11-17T01:20:45+01:00"
		  endTime:"2022-11-17T12:20:14+01:00"
      isClosed:false
      isDeliveryClosed:false
      isExpressDeliveryClosed:false
    }
  ]) {
        error
        errors
        scheduleExceptions {
          id
          isClosed
          isDeliveryClosed
	        startTime
          endTime
         }
        }
      }`;
  // openInGraphiql(query);
  const result = await fetchGraphQL(query);
  expect(result).toMatchSnapshot();
});
