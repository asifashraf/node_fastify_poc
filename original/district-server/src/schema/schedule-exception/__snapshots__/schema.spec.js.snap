// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`brand locations can resolve scheduleException 1`] = `
Object {
  "data": Object {
    "brandLocation": Object {
      "scheduleExceptions": Array [
        Object {
          "endTime": "2017-12-04T11:00:00Z",
          "id": "06c63b24-4a10-4a1c-8ffa-83d88803b147",
          "isClosed": true,
          "isDeliveryClosed": true,
          "startTime": "2017-12-04T03:00:00Z",
        },
        Object {
          "endTime": "2017-12-05T11:00:00Z",
          "id": "82d8762b-95af-48c9-ab92-bf45ce49bc1c",
          "isClosed": true,
          "isDeliveryClosed": true,
          "startTime": "2017-12-05T09:00:00Z",
        },
        Object {
          "endTime": "2017-12-06T05:00:00Z",
          "id": "ab818e56-3585-405d-8b30-51601f29e2b5",
          "isClosed": true,
          "isDeliveryClosed": true,
          "startTime": "2017-12-06T03:00:00Z",
        },
        Object {
          "endTime": "2017-12-07T09:00:00Z",
          "id": "2dd75f52-1b0e-4cea-b2ac-e28e7409dc20",
          "isClosed": true,
          "isDeliveryClosed": true,
          "startTime": "2017-12-07T07:00:00Z",
        },
        Object {
          "endTime": "2017-12-08T13:00:00Z",
          "id": "a2f8e7d6-8d51-4297-9cb7-1212ff946e42",
          "isClosed": false,
          "isDeliveryClosed": false,
          "startTime": "2017-12-08T11:00:00Z",
        },
      ],
    },
  },
}
`;

exports[`scheduleExceptions can be added 1`] = `
Object {
  "data": Object {
    "brandLocationScheduleExceptionSave": Object {
      "error": null,
      "errors": null,
      "scheduleExceptions": Array [
        Object {
          "endTime": "2022-11-16T11:20:14Z",
          "id": "0bbf9978-7442-4af6-941c-c981d69f4712",
          "isClosed": false,
          "isDeliveryClosed": false,
          "startTime": "2021-11-16T00:20:45Z",
        },
        Object {
          "endTime": "2022-11-17T11:20:14Z",
          "id": "c02b49ce-e934-4c84-992d-8383f6f5f13b",
          "isClosed": false,
          "isDeliveryClosed": false,
          "startTime": "2021-11-17T00:20:45Z",
        },
      ],
    },
  },
}
`;
